app/controllers/turbo/frames/frame_request.rb

```diff
@@ -9,6 +9,20 @@
 # Turbo Frames knows how to fish out the relevant frame regardless.
 #
 # This module is automatically included in <tt>ActionController::Base</tt>.
+#
+# You can override this behavior in <tt>ApplicationController</tt> if you want to have a layout for turbo frame requests:
+#
+#   # app/controllers/articles_controller.rb
+#   class ApplicationController < ActionController::Base
+#     layout -> { "application" if turbo_frame_request? }
+#   end
+#
+# Then, you have to create a layout with the corresponding name. It can be handy for flash messages, for example:
+#
+#   # app/views/layouts/application.turbo_stream.erb
+#   <%= turbo_stream.update('flash', partial: 'layouts/flash') %>
```

Comment:
Thank you for proposing this documentation!

The PR's title and the preceding code sample mention Frames, while the layout code sample mentions the `turbo_stream` helper and the template's file extension mentions `turbo_stream.erb`.

As I understand them, Streams and Frames are unrelated (but complementary) tools.

I think the first paragraph could be related to https://github.com/hotwired/turbo-rails/pull/232#issuecomment-1133901506, and feels appropriate in this file.

Is there another, more appropriate area of the codebase we could outline the Streams layout pattern?

Reply:
@seanpdoyle, thank you very much for your feedback! 

You are right, *turbo frame requests* and *turbo stream requests* are two different things. In fact, in most cases, we don't want the layout for *turbo frame requests* that are not *turbo stream requests*, as we are only going to extract and replace one single turbo frame. 

However, we want to have this `application.turbo_stream.erb` layout in the case of a *turbo stream request* for flash messages. Maybe we could add a method to identify *turbo stream requests* like this: 

```rb
def turbo_stream_request?
  request.accept.starts_with?(Mime[:turbo_stream])
end
```

Then we could only remove the layout for *turbo frame requests* that are not *turbo stream requests*:

```rb
layout -> { (turbo_frame_request? && !turbo_stream_request?) ? false : "application" }
```

Does it make sense? I just added a new commit so we can discuss this, let me know what you think! If we agree on this, maybe we can then improve the documentation even further.

Reply:
Thank you for clarifying @alexandreruban.

I was under the impression that [a `app/views/layouts/application.turbo_stream.erb` layout is already supported] (https://github.com/hotwired/turbo-rails/issues/25#issuecomment-815774859). Has that changed?

Reply:
@seanpdoyle sorry for the late reply. 

The `application.turbo_stream.erb` layout is supported when we override the `layout -> { false if turbo_frame_request? }` of the `turbo-rails` gem with `layout "application"` in our `ApplicationController` for example.

However, this has an unintended side effect. Now, when making GET requests by clicking links inside a Turbo Frame, the view is rendered with the `application.html.erb` layout. We don't need any layout in that case because Turbo is going to replace only the content of a specific Frame.

In fact, what we often want is to add an `application.turbo_stream.erb` layout for Turbo Stream requests (for flash messages most of the time) but not when we make Turbo Frame requests that are not Turbo Stream requests.

I created a [reproduction repository](https://github.com/alexandreruban/layout-reproduction) to clearly show what I mean. The description each commit is very precise and will show you precisely the issue.

Reply:
@alexandreruban thank you for sharing that reproduction repository. 
I think I understand the issue now. I really appreciate the granularity of the commits, it helped me tell the difference between the code generated by Rails and the code that was pertinent to the issue.